examples/moist_bf_bubble.py

```diff
@@ -0,0 +1,196 @@
+R_d = params.R_d
+R_v = params.R_v
+cp = params.cp
+c_pl = params.c_pl
+c_pv = params.c_pv
+L_v0 = params.L_v0
+kappa = params.kappa
+w_sat1 = params.w_sat1
+w_sat2 = params.w_sat2
+w_sat3 = params.w_sat3
+w_sat4 = params.w_sat4
+T_0 = params.T_0
+g = params.g
+cp = params.cp
+
+# Define constant theta_e and water_t
+Tsurf = 320.0
+total_water = 0.02
+theta_e = Function(Vt).assign(Tsurf)
+water_t = Function(Vt).assign(total_water)
+
+# Calculate hydrostatic fields
+moist_hydrostatic_balance(state, theta_e, water_t)
+
+# make mean fields
+theta_b = Function(Vt).assign(theta0)
+rho_b = Function(Vr).assign(rho0)
+water_vb = Function(Vt).assign(water_v0)
+water_cb = Function(Vt).assign(water_t - water_vb)
+
+# define perturbation
+xc = 10000.
+zc = 2000.
+rc = 2000.
+Tdash = 2.0
+theta_pert = Function(Vt).interpolate(conditional(sqrt((x[0] - xc) ** 2.0 + (x[1] - zc) ** 2.0) > rc,
+                                                  0.0, Tdash *
+                                                  (cos(pi * sqrt(((x[0] - xc) / rc) ** 2.0 + ((x[1] - zc) / rc) ** 2.0) / 2.0))
+                                                  ** 2.0))
+
+# define initial theta
+theta0.assign(theta_b * (theta_pert / 300.0 + 1.0))
+
+# find perturbed rho
+gamma = TestFunction(Vr)
+rho_trial = TrialFunction(Vr)
+a = gamma * rho_trial * dxp
+L = gamma * (rho_b * theta_b / theta0) * dxp
+rho_problem = LinearVariationalProblem(a, L, rho0)
+rho_solver = LinearVariationalSolver(rho_problem)
+rho_solver.solve()
+
+# find perturbed water_v
+w_v = Function(Vt)
+phi = TestFunction(Vt)
+
+p = p_0 * (R_d * theta0 * rho0 / p_0) ** (1.0 / (1.0 - kappa))
+T = theta0 * (R_d * theta0 * rho0 / p_0) ** (kappa / (1.0 - kappa)) / (1.0 + w_v * R_v / R_d)
+w_sat = w_sat1 / (p * exp(w_sat2 * ((T - T_0) / (T - w_sat3))) - w_sat4)
+
+w_functional = (phi * w_v * dxp - phi * w_sat * dxp)
+w_problem = NonlinearVariationalProblem(w_functional, w_v)
+w_solver = NonlinearVariationalSolver(w_problem)
+w_solver.solve()
+
+water_v0.assign(w_v)
+water_c0.assign(water_t - water_v0)
+
+# initialise fields
+state.initialise([('u', u0),
+                  ('rho', rho0),
+                  ('theta', theta0),
+                  ('water_v', water_v0),
+                  ('water_c', water_c0)])
+state.set_reference_profiles([('rho', rho_b),
+                              ('theta', theta_b),
+                              ('water_v', water_vb)])
+
+# Set up advection schemes
+ueqn = EulerPoincare(state, Vu)
+rhoeqn = AdvectionEquation(state, Vr, equation_form="continuity")
+thetaeqn = EmbeddedDGAdvection(state, Vt, equation_form="advective")
+
+advected_fields = [('u', ThetaMethod(state, u0, ueqn)),
+                   ('rho', SSPRK3(state, rho0, rhoeqn)),
+                   ('theta', SSPRK3(state, theta0, thetaeqn)),
+                   ('water_v', SSPRK3(state, water_v0, thetaeqn)),
+                   ('water_c', SSPRK3(state, water_c0, thetaeqn))]
+
+# Set up linear solver
+schur_params = {'pc_type': 'fieldsplit',
```

Comment:
I think you probably don't want these solver options - the default ones are better.